numpy/core/fromnumeric.py

```diff
@@ -993,7 +1001,11 @@ def sort(a, axis=-1, kind=None, order=None):
         axis = -1
     else:
         a = asanyarray(a).copy(order="K")
-    a.sort(axis=axis, kind=kind, order=order)
+
+    if by is not None and not isinstance(by, (tuple, list)):
+        raise TypeError("\'by\' is expected to be of a sequence type.")
```

Comment:
This seems like it should be unnecessary, because the check should live in `ndarray.sort()`.

I am actually happy with this decision: I.e. force a `tuple()` right now, and think about generalizing later.  In which case I think we should go with explicitly tuple, and even disallow lists.

Reply:
Thanks Sebastian, will address these changes